### PR TITLE
Add ability to deliver assets from byocdn domain

### PR DESCRIPTION
**Pull request by sdmcraft**
Test URLs:
- Before: https://main--franklin-assets-selector--hlxsites.hlx.live/same-domain-delivery
- After: https://gems--franklin-assets-selector--hlxsites.hlx.live/same-domain-delivery

### CODE REVIEW

**Comment by sdmcraft** [comment]
> Do you think there's value in a reverse pattern as well? (where AEM Assets sidekick plugin has an additional configuration to specify the custom domain to include in the urls it adds to the doc, and the custom domain is swapped for delivery domain when the site is accessed from _.hlx._ domains)

That's a good idea indeed. It would mean less custom work, just specify the prod domain in AS config and the AS addon swaps the domains while copy pasting in Word doc itself. The assumption would be that you should only do this after you have setup the routing in byocdn to Polaris origin so that delivery urls with byocdn domain are able to deliver assets.

I think I can even omit swapping back to delivery urls as the asset delivery via byocdn would still work even for pages served from the internal hlx domains.

**Comment by actinium15** [comment]
> The assumption would be that you should only do this after you have setup the routing in byocdn to Polaris origin so that delivery urls with byocdn domain are able to deliver assets.

...and we can forego that assumption _if_ we implement a reverse of what this PR does (i.e., _until_ byocdn is configured to front Polaris OOTB domain, the swapping will continue).

In my mind that'd mean lookup of img/picture/video tags in render-blocking won't be needed on 'production' websites - not sure if it is micro/meaningless optimization, though.

>  can even omit swapping back to delivery urls as the asset delivery via byocdn would still work even for pages served from the internal hlx domains.

+1, subject to the assumption above - though my hunch is that getting customer to `implement the routing to Polaris domain in byocdn _before go-live/CDN-cutover to Franklin site_` might prove challenging.

**Comment by sdmcraft** [comment]
I am thinking slightly differently on this one. Basically have the same domain asset delivery as an "opt-in" else the default is as now i.e. assets get delivered for delivery-pXX-eYY.  
Else for swapping back, we need to identify which urls needs rewrite (probably doable by looking at pattern like `/adobe/assets/urn:aaid:aem..` ) and keep track of back to what (which delivery-pXX-eYY..) and also have some flag to indicate that byocdn is set and swap back is no longer needed.

>  though my hunch is that getting customer to implement the routing to Polaris domain in byocdn _before go-live/CDN-cutover to Franklin site_ might prove challenging
Yes, but I think we can't work around that since we are tackling this outside the the pipeline that generates the markup

File: tools/sidekick/config.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
       "id": "asset-library",
       "title": "AEM Assets Library",
       "environments": ["edit"],
-      "url": "https://experience.adobe.com/solutions/CQ-assets-selectors/static-assets/resources/franklin/asset-selector.html",
+      "url": "https://experience.adobe.com/solutions/CQ-assets-selectors/static-assets/resources/franklin/asset-selector.html?extConfigUrl=https%3A%2F%2Fhelix-asset-selector-configs.netlify.app%2Fconfig-gems.json",
```
[the remark this answers is not in the document]
Comment (sdmcraft) [reply tools/sidekick/config.json]: Using Polaris now for delivering configs and works great 👍

**Review by shrotia — APPROVED** [review]

**Comment by sdmcraft** [comment]
Closing this and handling this on the Asset selector end wherein the authoring flow, the url copied over to Word/Google doc can be configured to have site's byocdn domain.

**Pull request closed without merging**